### ISSUE TITLE
Slicing string around `len()` (expansion of FURB188 slice-to-remove-prefix-or-suffix)

### ISSUE THREAD

**Issue by opk12** [astral-sh/ruff#13396]
This old code slices a string, using the `len()` of the prefix (which is a string variable) and suffix (which is a literal). I don't know if it was a common idiom, but Python 3.9 [introduced](https://docs.python.org/3.9/library/stdtypes.html#str.removeprefix) the more readable alternatives, `str.removeprefix()` + `str.removesuffix()`. They have the advantage that they don't raise, if the string does not match the prefix / suffix.

This is related to [FURB188](https://docs.astral.sh/ruff/rules/slice-to-remove-prefix-or-suffix/) `slice-to-remove-prefix-or-suffix`.

**Source**
```
def path_to_qualified_name(path: str, project_dir: str) -> str:
    return path[len(project_dir):-len('.py')].replace('/', '.')
```

**Command**
`ruff check .  --select ALL --isolated --preview`
